lib/kube/proxy/server_test.go

```diff
@@ -46,6 +47,18 @@ import (
 	"github.com/gravitational/teleport/lib/utils"
 )
 
+func TestServeConfigureError(t *testing.T) {
+	srv := &TLSServer{Server: &http.Server{TLSConfig: &tls.Config{MinVersion: tls.VersionTLS12, CipherSuites: []uint16{}}}}
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	defer listener.Close()
+
+	err = srv.Serve(listener)
+	require.Error(t, err) // expected due to incompatible ciphers
+
+	require.True(t, srv.mu.TryLock()) // verify that lock was released despite error
```

Comment:
I guess this never would have really deadlocked in practice, because if `Serve` fails, the entire service is considered terminated and we don't try to reuse the locked mutex.

Given that, this is a reasonable way to get some test coverage.

Reply:
Yes, it would have required an attempt to reuse the TLSServer to Serve again (which we don't currently do).

This was just an informational finding, but I thought was still worth a quick PR for technical correctness